ArrtSource/App/AppWindow.cpp

```diff
@@ -376,3 +370,48 @@ void ArrtAppWindow::closeEvent(QCloseEvent* e)
 
     QMainWindow::closeEvent(e);
 }
+
+void ArrtAppWindow::CheckForNewVersion()
+{
+    QPointer<ArrtAppWindow> thisPtr = this;
+
+    using namespace web::http;
+    client::http_client client(L"https://api.github.com/repos/Azure/azure-remote-rendering-asset-tool/releases/latest");
+    client.request(methods::GET).then([thisPtr, this](const pplx::task<http_response>& previousTask)
+                                      {
+                                          try
+                                          {
+                                              auto response = previousTask.get();
+                                              if (response.status_code() == status_codes::OK && thisPtr)
+                                              {
+                                                  auto json = response.extract_json().get();
+                                                  QString latestVersion = QString::fromStdWString(json.at(L"tag_name").as_string());
+
+                                                  QMetaObject::invokeMethod(QApplication::instance(), [this, latestVersion]()
+                                                                            { OnCheckForNewVersionResult(latestVersion); });
+                                              }
+                                          }
+                                          catch (...)
+                                          {
+                                              // in case of any exception, don't show the dialog
+                                          }
+                                      });
+}
+
+void ArrtAppWindow::OnCheckForNewVersionResult(QString latestVersion)
```

Comment:
what is common Qt pattern? Pass QString by value?

Reply:
Indeed, QString uses refcounted storage AFAIK, so that passing by value is efficient, because this is very often needed when sending events.